SD-K9/src/app/components/outdoor-map/outdoor-map.component.spec.ts

```diff
@@ -2,42 +2,69 @@ import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { IonicModule } from '@ionic/angular';
 
 import { OutdoorMapComponent } from './outdoor-map.component';
+import { BuildingInfoComponent } from '../building-info/building-info.component';
 import { Geolocation } from '@ionic-native/geolocation/ngx';
+import { GeolocationMock } from '../../test-helpers/mock-geolocation';
 import { OutdoorRouteBuilder } from 'src/app/providers/outdoor-route-builder.service';
-
-var geolocate = require('mock-geolocation');
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 
 describe('OutdoorMapComponent', () => {
   let component: OutdoorMapComponent;
   let fixture: ComponentFixture<OutdoorMapComponent>;
 
-  let point = [54.980206086231, 82.898068362003];
-
-  class MockGeolocation extends Geolocation {
-    getCurrentPosition() {
-        return new Promise<any>(geolocate.use());
-    }
-
-  }
-
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-        declarations: [ OutdoorMapComponent ],
-        imports: [IonicModule.forRoot()],
+        declarations: [ OutdoorMapComponent,BuildingInfoComponent ],
+        imports: [IonicModule.forRoot(),BrowserAnimationsModule],
         providers: [
-            {provide: Geolocation, useClass: MockGeolocation},
+            {provide: Geolocation, useClass: GeolocationMock},
             OutdoorRouteBuilder
         ]
     }).compileComponents();
 
     fixture = TestBed.createComponent(OutdoorMapComponent);
     component = fixture.componentInstance;
-    geolocate.use();
-    geolocate.send({lat: point[0], lng: point[1]});
+    component.data = {id: 1, building: 'hall', floor: 8};
     fixture.detectChanges();
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 1000000;
   }));
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should initialize the map with user coordinates', () => {
```

Comment:
Just a suggestion, you can also verify that GoogleStore.storeMap has been called when the map is first initialized.
You can also verify that GoogleStore.updateGloogleMap has been called when there's a new route direction or the map has been reloaded.

Reply:
I can't spy on those functions since they are hidden behind a private variable. Might be a good thing to test down the line though